### PR TITLE
Back button

### LINKED ISSUE #114: Back button
The back button in the header needs to go to the previous screen

On the goals list page the back button should not be shown at all

### PR DESCRIPTION
**Pull request by shiryz**
Fixes #114 .

### CODE REVIEW

File: app/reducers/index.js [hunk 1]
```diff
@@ -10,6 +10,41 @@ const defaultState = {
   currentGoal: {},
 };
 
+export const backStep = (state) => {
+  const step = state.step;
+  const previousStep = state.previousStep;
+
+  switch(step) {
+  case steps.ADD_GOAL:
```
Comment (des-des) [inline app/reducers/index.js]: indentation

File: app/reducers/index.js [hunk 2]
```diff
@@ -10,6 +10,41 @@ const defaultState = {
   currentGoal: {},
 };
 
+export const backStep = (state) => {
+  const step = state.step;
+  const previousStep = state.previousStep;
```
Comment (des-des) [inline app/reducers/index.js]: You can use desructuring
```js
const { step, previousStep } = state
```
Just an FYI

File: tests/reducer-test.js [hunk 3]
```diff
@@ -19,6 +20,65 @@ const defaultState = {
...
+    step: steps.RATE_GOAL,
+    previousStep: steps.GOALS_LIST,
+  };
+  const stateFeedback = {
+    ...defaultState,
+    step: steps.FEEDBACK,
+    previousStep: steps.RATE_GOAL,
+  };
+
+  t.equal(backStep(stateGoalsList).step, steps.GOALS_LIST, `stays on same step
```
Comment (des-des) [inline tests/reducer-test.js]: okay these descriptions do not work, ie see below: `previous step should stay null` ->  assertion does not describe *what* makes this happen..

Comment (des-des) [inline tests/reducer-test.js]:
```js
let step, previousStep;
{ step, previousStep } = backStep(stateGoalsList)
t.equal
t.equal
{ step, previousStep } = backStep(stateAddGoal)
t.equal
t.equal 
...
```
I think rebinding variables in tests is okay

File: app/reducers/index.js (hunk 1 above)
Comment (emilyb7) [inline app/reducers/index.js]: eslint says it's good

**Pull request merged**